packages/fetchai/connections/ledger_api/connection.py

```diff
@@ -60,6 +60,7 @@ def __init__(
         self._dispatcher = _RequestDispatcher(self.loop)
 
         self.receiving_tasks: List[asyncio.Future] = []
+        self.task_to_request: Dict[asyncio.Future, Envelope] = {}
```

Comment:
we need this to retrieve the ledger id from a response. For that we need to store the associated request.